File: java/src/test/java/com/peshchuk/topjslibs/PageJsSourcesTest.java
package com.peshchuk.topjslibs;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.HashSet;
import org.junit.jupiter.api.Test;

public class PageJsSourcesTest {

  @Test
  public void call() {
    // Arrange
    final var expected = new HashSet<String>();
    expected.add("base.min;v35620016.js");
    expected.add("modules.min;v35620016.js");
    expected.add("webtrekk_cookieControl.min;v35620016.js");
    expected.add("webtk.min;v35620016.js");

    // Act
    final var actual = new PageJsSources(() -> Util.getHtmlFrom("test.de.html")).call();

    // Assert
    assertEquals(expected, actual);
  }

}

File: kotlin/src/test/kotlin/com/peshchuk/topjslibs/PageJsSourcesTest.kt
package com.peshchuk.topjslibs

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.newSingleThreadContext
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.test.resetMain
import kotlinx.coroutines.test.setMain
import kotlin.test.AfterTest
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertEquals

@Suppress("EXPERIMENTAL_API_USAGE")
internal class PageJsSourcesTest {

    private val mainThreadSurrogate = newSingleThreadContext("Test thread")

    @BeforeTest
    fun setUp() {
        Dispatchers.setMain(mainThreadSurrogate)
    }

    @AfterTest
    fun tearDown() {
        Dispatchers.resetMain() // reset main dispatcher to the original Main dispatcher
        mainThreadSurrogate.close()
    }

    @Test
    fun get() {
        runBlocking {
            launch(Dispatchers.Main) {
                // Arrange
                val expected = setOf(
                    "base.min;v35620016.js",
                    "modules.min;v35620016.js",
                    "webtrekk_cookieControl.min;v35620016.js",
                    "webtk.min;v35620016.js"
                )
                val pageJsSources = PageJsSources { Util.getHtmlFrom("test.de.html") }

                // Act
                val actual = pageJsSources.get()

                // Assert
                assertEquals(expected, actual)
            }
        }
    }
}
